tests/test_api_check_mixin.py

```diff
@@ -91,4 +91,4 @@ def test_scope_manager_check_works(self):
             api_check.test_tracer_scope_manager_activate()
 
         with self.assertRaises(AssertionError):
-            api_check.test_start_active_span_not_finish_on_close()
+            api_check.test_start_active_span_finish_on_close()
```

Comment:
> # no-op tracer doesn't have a ScopeManager implementation

why? I think a similar question was asked for Java. The Noop tracer is supposed to "just work" with the same instrumentation that works for any other tracer - it may not support certain things like baggage propagation, but it should never raise exceptions in cases where a normal tracer would work.

Reply:
Oh, I think this is a problem of the documentation not being clear enough - Essentially the no-op `Tracer` does have a `ScopeManager` implementation, but it's a no-op too - which means no *actual* propagation will be done - but no exceptions are thrown (those mentioned here are raised by the `opentracing/harness.py` module), i.e.

```python
   def test_start_active_span_parent(self):
      tracer = self.tracer()
        with tracer.start_active_span('Fry') as parent:
            with tracer.start_active_span('Farnsworth') as child:
                if self.check_scope_manager():
                    # Will raise an Exception here - only for the tests.
                    assert self.is_parent(parent.span, child.span)
```

If it makes sense to you, I will update the documentation to make this point clear ;)

Reply:
I guess I am confused why the test is designed to raise an exception...

Reply:
Oh, actually we are following there the convention used by the **baggage** part from the tests:

```python
    def test_baggage_check_works(self):
        api_check = APICompatibilityCheckMixin()
        setattr(api_check, 'tracer', lambda: Tracer())

        # no-op tracer does not store baggage, so the test with default
        # value of `check_baggage_values()` should fail.
        with self.assertRaises(AssertionError):
            api_check.test_span_baggage()
```

(Of course, for a full-formed `Tracer` instance, including the `basictracer`, these `AssertionError` are supposed to never happen ;) )